### ISSUE TITLE
Shelly RGBW2 - change icon/function?

### ISSUE THREAD

**Issue by HannesHomeAss** [mongoose-os-apps/shelly-homekit#1411]
Hi

I use RGBW2 for irrigation. each channel controls its own valve. is it possible to change the icon or function so HomeKit understands that it is irrigation? Thanks!

**Comment by markirb** [comment]
Hi,

are you using them as binary in/ouput?

Currently not possible, though it would not be hard to implement:
Add a new device mode: 4 Output with stateless switches, and then you could do exactly that.

**Comment by HannesHomeAss** [comment]
Not sure what you mean with binary?

yeah it would be great if this feature could be implemented just like you can choose garage door with the Shelly Plus 1 when flashing

**Comment by HannesHomeAss** [comment]
> Hi,
> 
> are you using them as binary in/ouput?
> 

Aah like on/off. Yes, I do. The valves I control
Is 24V. So only on or off. Even tough I could dim for the moment in HomeKit.

**Comment by lucassite120** [comment]
You could do it with HAA firmware:
https://github.com/RavenSystem/esp-homekit-devices/wiki/water-valve

**Comment by markirb** [comment]
https://github.com/mongoose-os-apps/shelly-homekit/tree/rgbw2_switch

I tried to build this quickly.
Please test:
[release-ShellyRGBW2.zip](https://github.com/user-attachments/files/16140361/release-ShellyRGBW2.zip)

**Comment by HannesHomeAss** [comment]
Hi

How do I update with this zip OTA ?
Usually I just use the URL.

Thanks!

**Comment by HannesHomeAss** [comment]
Tried:

￼

**Comment by HannesHomeAss** [comment]
Think I solved it, but new error:
￼

**Comment by HannesHomeAss** [comment]
<img width="475" alt="Skärmavbild 2024-07-10 kl  10 12 23" src="https://github.com/mongoose-os-apps/shelly-homekit/assets/121405352/70bf83bd-1ced-4d1f-96ea-202c99304f71">

**Comment by markirb** [comment]
Oh sorry the zip is within the zip

**Comment by HannesHomeAss** [comment]
Yes. Tried the zip in the zip. Doesn’t work 10 juli 2024 kl. 19:28 skrev Markus Kirberg ***@***.***>:﻿
Oh sorry the zip is within the zip

**Comment by markirb** [comment]
I see. Indeed the firmware is now too big... I have to see where I can take cuts

**Comment by markirb** [comment]
Can you try this here?

You have to set it to "Normal mode"
[shelly-rgbw2-switch.zip](https://github.com/user-attachments/files/16324503/shelly-rgbw2-switch.zip)
